api/app/scripts/disaster_recovery.py

```
@@ -0,0 +1,1709 @@
+        current /= part
+        if current.is_symlink():
+            raise RecoveryError("Extraction destination contains a symlink")
+        if current.exists():
+            if not current.is_dir():
+                raise RecoveryError("Extraction destination path is not a directory")
+            continue
+        try:
+            current.mkdir(mode=0o700)
+        except FileExistsError:
+            if current.is_symlink() or not current.is_dir():
+                raise RecoveryError("Extraction destination path is unsafe")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)